aten/src/ATen/native/Distributions.cpp

```diff
@@ -266,30 +267,75 @@ Tensor& normal_meta_(Tensor& self, double mean, double std, c10::optional<Genera
   return self;
 }
 
+// out tensor float
 Tensor& normal_out(const Tensor& mean, double std, c10::optional<Generator> gen, Tensor& output) {
   return at::native::templates::normal_out_impl<NormalStub, Generator>(output, mean, std, gen);
 }
 
+Tensor& normal_out_meta(const Tensor& mean, double std, c10::optional<Generator> gen, Tensor& output) {
+  CHECK_NORMAL_STD(std);
+  auto std_tensor = at::empty_like(output, MemoryFormat::Contiguous);
+  at::native::templates::resize_output_for_normal(output, mean, std_tensor);
+  return output;
+}
+
+// out float tensor
 Tensor& normal_out(double mean, const Tensor& std, c10::optional<Generator> gen, Tensor& output) {
   return at::native::templates::normal_out_impl<NormalStub, Generator>(output, mean, std, gen);
 }
 
+Tensor& normal_out_meta(double mean, const Tensor& std, c10::optional<Generator> gen, Tensor& output) {
+  CHECK_NORMAL_TENSOR_STD(std);
+  auto mean_tensor = at::empty_like(output, MemoryFormat::Contiguous);
+  at::native::templates::resize_output_for_normal(output, mean_tensor, std);
+  return output;
+}
+
+// out tensor tensor
 Tensor& normal_out(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen, Tensor& output) {
   return at::native::templates::normal_out_impl<NormalStub, Generator>(output, mean, std, gen);
 }
 
+Tensor& normal_out_meta(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen, Tensor& output) {
+  CHECK_NORMAL_TENSOR_STD(std);
+  at::native::templates::resize_output_for_normal(output, mean, std);
+  return output;
+}
+
+// functional tensor float
 Tensor normal(const Tensor& mean, double std, c10::optional<Generator> gen) {
   return at::native::templates::normal_impl<NormalStub, Generator>(mean, std, gen);
 }
 
+Tensor normal_meta(const Tensor& mean, double std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_STD(std);
+  auto ret = at::empty_like(mean, MemoryFormat::Contiguous);
+  return ret;
+}
+
+// functional float tensor
 Tensor normal(double mean, const Tensor& std, c10::optional<Generator> gen) {
   return at::native::templates::normal_impl<NormalStub, Generator>(mean, std, gen);
 }
 
+Tensor normal_meta(double mean, const Tensor& std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_TENSOR_STD(std);
+  auto ret = at::empty_like(std, MemoryFormat::Contiguous);
+  return ret;
+}
+
+// functional tensor tensor
 Tensor normal(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
   return at::native::templates::normal_impl<NormalStub, Generator>(mean, std, gen);
 }
 
+Tensor normal_meta(const Tensor& mean, const Tensor& std, c10::optional<Generator> gen) {
+  CHECK_NORMAL_TENSOR_STD(std);
+  auto ret = at::empty_like(mean, MemoryFormat::Contiguous);
```

Comment:
You might want to explicitly pass the meta device `(c10::kMeta)` into `empty_like()` here, just to be safe.

Technically, if you have mixed device inputs (say `mean` is on CPU, and `std` is on Meta), then we'll end up dispatching to this meta kernel. That happens to be true because "Meta" has higher priority than "CPU" in the dispatcher.

Reply:
just a comment for whoever find this: this was attempted at first, but after refactoring it's not possible to differentiate between meta/non-meta tensors anymore. so marking as resolved.